app/lib/common/widgets/emoji_picker_widget.dart

```diff
@@ -83,4 +102,140 @@ class EmojiPickerWidget extends StatelessWidget {
       ),
     );
   }
+
+  Widget actionBar(
+    BuildContext context,
+    EmojiViewConfig emojiConfig,
+    EmojiViewState state,
+    TabController tabController,
+    PageController pageController,
+  ) =>
+      Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          IconButton(
+            onPressed: state.onShowSearchView,
+            icon: Icon(PhosphorIcons.magnifyingGlass()),
+          ),
+          Expanded(
+            child: DefaultCategoryView(
+              Config(
+                emojiViewConfig: emojiConfig,
+                categoryViewConfig: CategoryViewConfig(
+                  backgroundColor: Theme.of(context).colorScheme.surface,
+                  initCategory: Category.RECENT,
+                ),
+              ),
+              state,
+              tabController,
+              pageController,
+            ),
+          ),
+          if (onBackspacePressed != null)
+            IconButton(
+              onPressed: onBackspacePressed,
+              icon: Icon(PhosphorIcons.backspace()),
+            ),
+          IconButton(
+            onPressed: onClosePicker,
+            icon: Icon(PhosphorIcons.xCircle()),
+          ),
+        ],
+      );
+}
+
+/// Default Search implementation
+class _CustomSearchView extends SearchView {
+  final VoidCallback closePicker;
+
+  /// Constructor
+  const _CustomSearchView(
+    super.config,
+    super.state,
+    super.showEmojiView,
+    this.closePicker,
+  );
+
+  @override
+  _CustomSearchViewState createState() => _CustomSearchViewState();
+}
+
+/// Default Search View State
+class _CustomSearchViewState extends SearchViewState<_CustomSearchView> {
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(builder: (context, constraints) {
+      final emojiSize =
+          widget.config.emojiViewConfig.getEmojiSize(constraints.maxWidth);
+      final emojiBoxSize =
+          widget.config.emojiViewConfig.getEmojiBoxSize(constraints.maxWidth);
+
+      return Container(
+        color: widget.config.searchViewConfig.backgroundColor,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Material(
+              color: Colors.transparent,
+              child: SizedBox(
+                height: emojiBoxSize + 8.0,
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: ListView.builder(
+                        padding: const EdgeInsets.symmetric(vertical: 4.0),
+                        scrollDirection: Axis.horizontal,
+                        itemCount: results.length,
```

Comment:
from where `results` array coming, facing difficulty to find it on code changes.

Reply:
this is an `extends SearchViewState<_CustomSearchView>`, so from its [original source](https://github.com/Fintasys/emoji_picker_flutter/blob/0978a59beb504474d334effd2a131d517ec673fd/lib/src/search_view/search_view.dart#L36) .

the entire code is just a copy-paste extended with the x-for-closing